docs/DeveloperGuide.md

```diff
@@ -509,17 +509,147 @@ testers are expected to do more *exploratory* testing.
+
+    2. Test case: `clear`<br>
+       Expected: A message stating that the app has been cleared shown in Result Display Panel.
+
+    3. Test case: `cler`<br>
+       Expected: List of Udders is not cleared. Error details of an unrecognised command shown in the Result Display Panel.
+
 ### Deleting an Udder
 
-1. Deleting an Udder while all udders are being shown
+1. Deleting an Udder while all Udders are being shown
+
+    1. Prerequisites: List all Udders using the `list` command. Multiple Udders in the list.
+
+    2. Test case: `delete 1`<br>
+       Expected: First Udder is deleted from the list. Details of the deleted Udder shown in the Result Display Panel.
+
+    3. Test case: `delete 0`<br>
+       Expected: No Udder is deleted. Error details shown in the status message.
 
-   1. Prerequisites: List all udders using the `list` command. Multiple udders in the list.
+    4. Other incorrect delete commands to try: `delete`, `delete x`, `...` (where x is larger than the list size)<br>
+       Expected: Similar to previous.
 
-   2. Test case: `delete 1`<br>
-      Expected: First udder is deleted from the list. Details of the deleted udder shown in the status message. Timestamp in the status bar is updated.
+### Editing an Udder
 
-   3. Test case: `delete 0`<br>
-      Expected: No udder is deleted. Error details shown in the status message. Status bar remains the same.
+1. Editing an Udder shown in the list
 
-   4. Other incorrect delete commands to try: `delete`, `delete x`, `...` (where x is larger than the list size)<br>
+   1. Prerequisites: List all Udders using the `list` command. Multiple Udders in the list.
+
+   2. Test case: `edit 1 p/91234567 e/johndoe@example.com`<br>
+      Expected: First Udder is edited. New details of the edited Udder shown in the Result Display Panel.
+
+   3. Test case: `edit 0 p/91234567 e/johndoe@example.com`<br>
+      Expected: No Udder is edited. Error details shown in the Result Display Panel.
+
+   4. Test case: `edit 1 p/00 e/johndoe@example.com`<br>
+      Expected: No Udder is edited. Error details regarding the phone number shown in the Result Display Panel.
+
+   5. Other incorrect edit commands to try: `edit`, `edit x`, `...` (where x is larger than the list size)<br>
       Expected: Similar to previous.
+
+### Finding an Udder
+
+1. Finding an Udder from an unfiltered list of Udders
+
+    1. Prerequisites: List all Udders using the `list` command. Multiple Udders in the list.
+
+    2. Test case: `find n/John`<br>
+       Expected: All Udders that contain the word "John" in the name is listed in the Person List Panel.
+
+    3. Test case: `find John`<br>
+       Expected: No filtering is done. Error details shown in the Result Display Panel.
+
+    4. Other incorrect delete commands to try: `find`, `find p/`, `...`<br>
+       Expected: Similar to previous.
+
+### Listing all Udders
+
+1. Listing all Udders from a filtered list of Udders
```

Comment:
Might this be misleading, do you mean `Listing all Udders after performing `find` command`